packages/backend-common/src/appContext/ApplicationContext.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import { Container, interfaces } from 'inversify';
+import express from 'express';
+import { AnyDependencyConfig, Dependency } from '@backstage/app-context-common';
+
+export interface ApplicationContext {
```

Comment:
Because it's gonna take some time to dive in an understand this I wanna throw in a preemptive comment here. I'll be digging into this interface and the other ones and looking for anything that is unnecessary and that could be removed 🔍